tests/ui/redundant_pattern_matching.stderr

```diff
@@ -1,31 +1,31 @@
 error: redundant pattern matching, consider using `is_ok()`
-  --> $DIR/redundant_pattern_matching.rs:5:12
+  --> $DIR/redundant_pattern_matching.rs:6:12
    |
 LL |     if let Ok(_) = Ok::<i32, i32>(42) {}
-   |     -------^^^^^------------------------ help: try this: `if Ok::<i32, i32>(42).is_ok()`
+   |     -------^^^^^------------------------ help: try this: `Ok::<i32, i32>(42).is_ok()`
    |
    = note: `-D clippy::redundant-pattern-matching` implied by `-D warnings`
 
 error: redundant pattern matching, consider using `is_err()`
-  --> $DIR/redundant_pattern_matching.rs:7:12
+  --> $DIR/redundant_pattern_matching.rs:8:12
    |
 LL |     if let Err(_) = Err::<i32, i32>(42) {}
-   |     -------^^^^^^------------------------- help: try this: `if Err::<i32, i32>(42).is_err()`
+   |     -------^^^^^^------------------------- help: try this: `Err::<i32, i32>(42).is_err()`
```

Comment:
Well now this is semantically wrong, since the `if let` returns the unit type `()` and the suggestion a bool. Is there an easy way to differentiate between those two cases and in one case suggest `if _.is_none() {}` and in the other `_.is_none()`?

Reply:
I'd much rather fix the problems in a separate PR. This is now already marked as `MaybeIncorrect` so we can focus on fixing other low-hanging fruit before `cargo fix --clippy` becomes available on stable.

I'll create an issue for the improvements.

Reply:
Ok, I added the comment, for both fixes, that need to be made for this to be `MachineApplicable`

Reply:
Thanks! Removing it from my todo list then :sweat_smile: